### PR TITLE
MLIBZ-1974: log time elapsed in a few key methods

### PR DESCRIPTION
**Pull request by heyzooi**
#### Description

Log time elapsed for performance measure

#### Changes

* Add log statements in a few key methods

#### Tests

* No need

### CODE REVIEW

File: Kinvey/Kinvey/FindOperation.swift [hunk 1]
```diff
@@ -164,7 +164,13 @@ internal class FindOperation<T: Persistable>: ReadOperation<T, [T], Swift.Error>
                         self.executeLocal(completionHandler)
                     }
                 } else {
-                    let entities = [T](JSONArray: jsonArray)
+                    func convert(_ jsonArray: [JsonDictionary]) -> [T] {
+                        let startTime = CFAbsoluteTimeGetCurrent()
+                        let entities = [T](JSONArray: jsonArray)
+                        log.debug("Time elapsed: \(CFAbsoluteTimeGetCurrent() - startTime) s")
```
Comment (tejasranade) [inline Kinvey/Kinvey/FindOperation.swift]: @heyzooi how do we know what method this "time elapsed" comes from? Is there any info in the debug logs indicating that this is from JSON conversion in the `FindOperation`?

File: Kinvey/Kinvey/HttpRequest.swift [hunk 2]
```diff
@@ -343,9 +343,7 @@ internal class HttpRequest: TaskProgressRequest, Request {
         let url = endpoint.url
         request = URLRequest(url: url)
         request.httpMethod = httpMethod.stringValue
-        if let timeout = options?.timeout {
-            request.timeoutInterval = timeout
-        }
+        request.timeoutInterval = options?.timeout ?? client.options?.timeout ?? client.timeoutInterval
```
Comment (tejasranade) [inline Kinvey/Kinvey/HttpRequest.swift]: @heyzooi this looks odd. If `options.timeout` is provided, shouldn't that be used?

File: Kinvey/Kinvey/HttpResponse.swift [hunk 3]
```diff
@@ -57,6 +57,13 @@ struct HttpResponse: Response {
     var etag: String? {
         return response.allHeaderFields["Etag"] as? String
     }
+    
+    var contentTypeIsJson: Bool {
```
Comment (tejasranade) [inline Kinvey/Kinvey/HttpResponse.swift]: @heyzooi this method doesn't seem to be called anywhere. Is it necessary?

File: Kinvey/Kinvey/LocalResponse.swift [hunk 4]
```diff
@@ -18,5 +18,6 @@ class LocalResponse: Response {
     var isMethodNotAllowed = false
     
     var etag: String? = nil
+    var contentTypeIsJson: Bool = false
```
Comment (tejasranade) [inline Kinvey/Kinvey/LocalResponse.swift]: Is this used?

File: Kinvey/KinveyMacAppTests/PerformanceTest.swift [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+        price <- ("price", map["Price"])
+        currency <- ("currency", map["Currency"])
+        deliveryUnit <- ("deliveryUnit", map["DeliveryUnit"])
+        unitQuantity <- ("unitQuantity", map["UnitQuantity"])
+        unitOfMeasure <- ("unitOfMeasure", map["UnitOfMeasure"])
+    }
+    
+}
+
+class PerformanceTest: XCTestCase {
```
Comment (tejasranade) [inline Kinvey/KinveyMacAppTests/PerformanceTest.swift]: @heyzooi this should not be run as part of our usual suite, fyi :)

File: Kinvey/Kinvey/FindOperation.swift (hunk 1 above)
Comment (heyzooi) [inline Kinvey/Kinvey/FindOperation.swift]: The log will include the file, the line number and the name of the method (even for internal methods like this) where the message was logged

File: Kinvey/Kinvey/HttpRequest.swift (hunk 2 above)
Comment (heyzooi) [inline Kinvey/Kinvey/HttpRequest.swift]: yes, indeed is begin used... the ?? operator try to use the value in the left side first, if there's no value then it uses the value in the right side

File: Kinvey/Kinvey/HttpResponse.swift (hunk 3 above)
Comment (heyzooi) [inline Kinvey/Kinvey/HttpResponse.swift]: nope, this was one of the changes that I made to try a few things, going to remove that

File: Kinvey/Kinvey/LocalResponse.swift (hunk 4 above)
Comment (heyzooi) [inline Kinvey/Kinvey/LocalResponse.swift]: nope, I will remove

File: Kinvey/KinveyMacAppTests/PerformanceTest.swift (hunk 5 above)
Comment (heyzooi) [inline Kinvey/KinveyMacAppTests/PerformanceTest.swift]: sure, and I added to our test suite, this was just a test

Comment (tejasranade) [inline Kinvey/KinveyMacAppTests/PerformanceTest.swift]: Just making sure that it's *not* part of our standard test suite.

**Pull request merged**